Test infra: conftest.py blocks pytest collection of tests/unit/marine_safety/

## Context

Discovered during PR #324 verification (part of #322 mop-up).

## The problem

Running pytest against any file under `tests/unit/marine_safety/` fails at **collection time** with:

\`\`\`
ImportError while loading conftest '/.../tests/unit/marine_safety/conftest.py'.
.../tests/unit/marine_safety/conftest.py:16: in <module>
    from tests.modules.marine_safety.fixtures.sample_data import (
E   ModuleNotFoundError: No module named 'tests.modules.marine_safety.fixtures'
\`\`\`

The conftest imports fixtures from a path that doesn't exist. Effect: **the entire `tests/unit/marine_safety/` directory is uncollectable**, not just one test — which means any future PR touching marine_safety tests can't verify via pytest.

## How to fix

Two plausible paths, pick whichever matches project intent:

1. **Restore the missing fixtures module** at `tests/modules/marine_safety/fixtures/sample_data.py`. Check git history:
   \`\`\`bash
   git log --all --full-history --oneline -- tests/modules/marine_safety/fixtures/sample_data.py
   \`\`\`
   If it existed and was removed, restore it.

2. **Rewrite the conftest** to use an alternate fixture source (inline fixtures, a simpler path, or factories). Update `conftest.py:16` accordingly.

Either fix should be accompanied by a smoke test: `uv run --extra dev pytest tests/unit/marine_safety/ --collect-only` should succeed with non-zero collected items.

## Refs

Surfaced during #324 but pre-existing (not introduced by that PR). Blocks downstream runtime validation of #324's marine_safety work. Low-urgency since #324 verified via flake8 static analysis, but should be fixed before the next marine_safety-adjacent PR.

--- a/tests/unit/marine_safety/conftest.py
+++ b/tests/unit/marine_safety/conftest.py
@@ -13,7 +13,7 @@ from sqlalchemy import create_engine, event
 from sqlalchemy.orm import sessionmaker
 from sqlalchemy.pool import StaticPool
 
-from tests.modules.marine_safety.fixtures.sample_data import (
+from tests.unit.marine_safety.fixtures.sample_data import (
     SAMPLE_INCIDENT_DATA,
     SAMPLE_USCG_PDF_TEXT,
     SAMPLE_USCG_REPORT,

--- a/tests/unit/marine_safety/fixtures/__init__.py
+++ b/tests/unit/marine_safety/fixtures/__init__.py
@@ -4,7 +4,7 @@ This package provides sample data generators and fixtures for testing
 marine safety incident database operations and parsers.
 """
 
-from tests.modules.marine_safety.fixtures.sample_data import (
+from tests.unit.marine_safety.fixtures.sample_data import (
     SAMPLE_INCIDENT_DATA,
     SAMPLE_USCG_PDF_TEXT,
     SAMPLE_USCG_REPORT,

